### PR TITLE
Add graphql support

### PR DESCRIPTION
**Pull request by tb**

### CODE REVIEW

File: src/assets/GraphqlAsset.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const Asset = require('../Asset');
+const gql = require('graphql-tag');
```
Comment (devongovett) [inline src/assets/GraphqlAsset.js]: this should be a local dependency in the project. you can use the localRequire util function to require relative to the project directory. see here for an example: https://github.com/parcel-bundler/parcel/blob/master/src/assets/SASSAsset.js#L10

File: test/javascript.js [hunk 2]
```diff
@@ -248,4 +249,32 @@ describe('javascript', function() {
     assert.equal(typeof output, 'function');
     assert.equal(output(), 3);
   });
+
+  it('should support requiring graphql files', async function() {
```
Comment (devongovett) [inline test/javascript.js]: can you move this to a separate test file for graphql?

**Comment by brandon93s** [comment]
@tb Looks like there is just a few feedback items. Have capacity to tackle them? Otherwise, looks ready to land as a newly supported asset type!

File: test/javascript.js (hunk 2 above)
Comment (tb) [reply test/javascript.js]: Ok, I will move

File: src/assets/GraphqlAsset.js (hunk 1 above)
Comment (tb) [reply src/assets/GraphqlAsset.js]: Ok, will use that

**Comment by tb** [comment]
@brandon93s @devongovett PR is rebased, feedback applied

File: src/assets/GraphqlAsset.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const localRequire = require('../utils/localRequire');
+
+class GraphqlAsset extends Asset {
+  constructor(name, pkg, options) {
+    super(name, pkg, options);
+    this.type = 'js';
+  }
+
+  async parse(code) {
+    let gql = await localRequire('graphql-tag', this.name);
```
Comment (devongovett) [inline src/assets/GraphqlAsset.js]: Question: does this work if `graphql` and `graphql-tag` are not already installed? In that case `graphql-tag` would be autoinstalled by parcel, but not `graphql` which is a peer dependency. Would that cause it to break?

Perhaps we will need to extend the autoinstall mechanism to also install peer dependencies... Or I guess we could just `localRequire('graphql')` here as well, which would force an install.

Comment (brandon93s) [reply src/assets/GraphqlAsset.js]: @devongovett Ah, good catch. No, it won't work if they aren't already installed. Fortunately, the error message will be along the lines of "cannot find `graphql`". My preference is to enhance auto install to check and handle peer dependencies. 

Adding `localRequire('graphql')` would be an unclear side-effect in the code. If `graphql-tag` ever drops it as a peer dependency, we'd keep installing it for no reason.

Comment (devongovett) [reply src/assets/GraphqlAsset.js]: Ok, let's do that as a separate PR.

**Pull request merged**